File: src/app/core/models/teacher.models.ts
/**
 * Teacher-related models
 * Extracted from teacher.service.ts and shared.models.ts
 */

export interface TeacherSubject {
    id: string;
    name: string;
    medium: 'Sinhala' | 'Tamil' | 'English';
    level: 'Primary' | 'OLevel' | 'ALevel' | 'Secondary' | 'Advanced';
}

export interface TeacherAvailability {
    dayOfWeek: string;
    startTime: string;
    endTime: string;
}

export interface TeacherProfile {
    id: string;
    userId: string;
    fullName: string;
    email: string;
    phoneNumber: string;
    profilePicture?: string;
    bio?: string;
    qualifications: string[];
    subjects: TeacherSubject[];
    hourlyRate: number;
    experienceYears: number;
    averageRating: number;
    totalReviews: number;
    totalClasses: number;
    isAvailable: boolean;
    availability: TeacherAvailability[];
    verificationStatus: 'Pending' | 'Verified' | 'Rejected';
    createdAt: Date;
    updatedAt: Date;
}

export interface UpdateTeacherRequest {
    fullName?: string;
    phoneNumber?: string;
    bio?: string;
    qualifications?: string[];
    subjects?: TeacherSubject[];
    hourlyRate?: number;
    experienceYears?: number;
    availability?: TeacherAvailability[];
    isAvailable?: boolean;
}

File: Backend/ClassBooking.API/Models/TeacherModels.cs
namespace ClassBooking.API.Models
{
    public class TeacherProfile
    {
        public string Id { get; set; } = string.Empty;
        public string UserId { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string? ProfilePicture { get; set; }
        public string? Bio { get; set; }
        public List<string> Qualifications { get; set; } = new();
        public List<TeacherSubject> Subjects { get; set; } = new();
        public decimal HourlyRate { get; set; }
        public int ExperienceYears { get; set; }
        public double AverageRating { get; set; }
        public int TotalReviews { get; set; }
        public int TotalClasses { get; set; }
        public bool IsAvailable { get; set; }
        public List<TeacherAvailability> Availability { get; set; } = new();
        public string VerificationStatus { get; set; } = "Pending"; // Pending, Verified, Rejected
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    public class TeacherSubject
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Medium { get; set; } = "English"; // Sinhala, Tamil, English
        public string Level { get; set; } = "OLevel"; // Primary, OLevel, ALevel, Secondary, Advanced
    }

    public class TeacherAvailability
    {
        public string DayOfWeek { get; set; } = string.Empty;
        public string StartTime { get; set; } = string.Empty;
        public string EndTime { get; set; } = string.Empty;
    }
}
